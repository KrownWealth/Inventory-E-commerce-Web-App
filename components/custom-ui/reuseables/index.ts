import InputSearch from "./Search/searchBar";
import SideBarNav from "./Navigation/sideBar";
import UserProfile from "./Profile/userProfileDropdown";
import { BreadcrumbWithCustomSeparator } from "./Navigation/breadCrumb";
import ProductPageHead from "./Headings/productPageHead";
import ModalManager from "./Modals/handleModalOpenClose";
import PageHead from "./Headings/PageHeading";
import PriceFormulaModal from "./Modals/salesPriceFormula";
import CustomerTable from "./Admin/customersTable";
import OrderStatusBtn from "./Buttons/orderStatusBtn";
import OrdersTable from "./Admin/ordersTable";
import { DatePickerWithRange } from "./DateRange/dateRangePicker";
import UserAvatar from "./Profile/userAvatar";
import DashBoardCard from "./Cards/dashBoardCard";
import SalesBarCharts from "./Admin/salesBarCharts";
import SalesCategoryDoughnut from "./Admin/salesCategoryouhnutCharts";
import CustomerMap from "./Admin/customerMapLineChart";
import ProductTableTwo from "./Admin/productTableTwo";
import { ProductCardSkeleton } from "./Skeleton/productTableSkeleton";
import ProductsPagination from "./Pagination/productPagination";


import SignupWithGoogle from "./Authentication/google0Auth";

import LanguagSelect from "./Localization/languagesSelect";
import {Quote} from "./Authentication/quote";
import { LogoOnAuth } from "./Navigation/logoOnAuth";
import { WelcomeText } from "./Headings/welcomeText";

import { UserAvatarDropdown } from "./Navigation/avatarDropdown";
import { ProductCard } from "./Cards/ProductCard";
import UserSearch from "./Search/userSearchBar";
import AddCategoryModal from "./Modals/addCategoryModal";
import CategoryTable from "./Admin/categoryTable";



export { InputSearch, SideBarNav, UserProfile, BreadcrumbWithCustomSeparator, 
  ProductPageHead, ModalManager, PageHead, PriceFormulaModal, CustomerTable,
OrderStatusBtn, DatePickerWithRange, UserAvatar, DashBoardCard, SalesBarCharts, SalesCategoryDoughnut,
CustomerMap, ProductTableTwo, ProductCardSkeleton, ProductsPagination, OrdersTable,  LanguagSelect, CategoryTable,
SignupWithGoogle, Quote, LogoOnAuth, WelcomeText, UserAvatarDropdown, ProductCard, UserSearch, AddCategoryModal }