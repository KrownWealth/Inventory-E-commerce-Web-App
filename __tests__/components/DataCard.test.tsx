import { render, screen } from '@testing-library/react';
import { DataCard } from '@/components/custom-ui/reuseables';
import { vi } from 'vitest';

describe('DataCard', () => {
  const mockProps = {
    dataHeading: 'Total Sales',
    dataContent: '500',
    description: 'Total sales made this month.',
    icon: <span>ðŸ’°</span>,
  };

  it('should renders the DataCard with correct props', () => {
    render(<DataCard {...mockProps} />);

    const paragraphs = screen.getAllByRole('paragraph');
    const heading = screen.getByRole('heading');

    expect(heading).toBeInTheDocument();

    expect(heading).toHaveTextContent(/Total/i);


    expect(paragraphs).toHaveLength(3);
    expect(paragraphs[0]).toHaveTextContent(/Total Sales/i);
    expect(paragraphs[1]).toHaveTextContent(/500/i);
    expect(paragraphs[2]).toHaveTextContent(/Total sales made this month./i);

    expect(screen.getByText(/ðŸ’°/i)).toBeInTheDocument();
  });


  it('should renders with the correct structure', () => {
    const { container } = render(<DataCard {...mockProps} />);
    expect(container.firstChild).toHaveClass('w-full');
    expect(container).toMatchSnapshot();
  });
});
