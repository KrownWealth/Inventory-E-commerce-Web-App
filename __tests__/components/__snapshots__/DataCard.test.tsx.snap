// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataCard > should renders with the correct structure 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full"
  >
    <div
      class="p-3 flex items-center space-x-2"
    >
      <span>
         
        <span>
          ðŸ’°
        </span>
      </span>
      <p
        class="text-muted-foreground text-sm"
      >
        Total Sales
      </p>
    </div>
    <div
      class="flex flex-col space-y-2 p-3"
    >
      <div
        class="flex justify-between "
      >
        <h4
          class="text-2xl font-bold"
        >
          Total
        </h4>
        <p>
          500
        </p>
      </div>
      <p>
        Total sales made this month.
      </p>
    </div>
  </div>
</div>
`;
